lib/charms/prometheus_k8s/v1/prometheus_remote_write.py

```diff
@@ -453,6 +456,9 @@ def __init__(
         self.framework.observe(
             self._charm.on.upgrade_charm, self._push_alerts_to_all_relation_databags
         )
+        self.framework.observe(
+            self._charm.on.config_changed, self._push_alerts_to_all_relation_databags
+        )
```

Comment:
Do we need to observe config-changed in the scrape lib as well? Or, do we need to observe config_changed here? This information coming from a config option is a charm impl detail we shouldn't leak.

Reply:
It's only necessary in Prometheus remote-write. For scrape, alerts end up in Prometheus' unit data; for remote-write, it's in the other charm's unit data. We're already observing **leader-elected** and **upgrade-charm**, I don't think this changes much.

Changing to a generic `refresh_event` like we have in other libraries :)